CHANGELOG.md

```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 4.0.0
```

Comment:
Do we need to increment the major version? I don't see any changes in the public API of the library. Perhaps incrementing the minor version would be more appropriate. What do you think?